Section: ED-EMS application

MichalisPanayides opened the pull request.

MichalisPanayides committed:
Add ARD with PoA plots in imgs/ directory

Each added pdf has four plots on it:
- Asymmetric replicator dynamics stratgies for row player
- Asymmetric replicator dynamics stratgies for column player
- Blocking time price of anarchy over time for row player
- Blocking time price of anarchy over time for column player

MichalisPanayides committed:
Add ARD with PoA plots in imgs/ directory

Each added pdf has four plots on it:
- Asymmetric replicator dynamics stratgies for row player
- Asymmetric replicator dynamics stratgies for column player
- Blocking time price of anarchy over time for row player
- Blocking time price of anarchy over time for column player

MichalisPanayides committed:
Add initial draft of Application section

MichalisPanayides committed:
Add initial draft of Application section

MichalisPanayides committed:
Update known words

MichalisPanayides committed:
Update known words

MichalisPanayides committed:
Add info about asymmetric replicator dynamics

MichalisPanayides committed:
Add info about asymmetric replicator dynamics

MichalisPanayides committed:
Improve asymmetric rd plots

Increase the granularity of the PoA curve so that it looks nicer

MichalisPanayides committed:
Improve asymmetric rd plots

Increase the granularity of the PoA curve so that it looks nicer

MichalisPanayides committed:
Add jupyter notebook for the base case of ard run

The jupyter notebook shows some additional details of the generated
plot.

MichalisPanayides committed:
Add jupyter notebook for the base case of ard run

The jupyter notebook shows some additional details of the generated
plot.

MichalisPanayides committed:
Fix figure labels and include some citations

MichalisPanayides committed:
Fix figure labels and include some citations

MichalisPanayides committed:
Add extra checks (#12)

* Update environment to include proselint

* Add proselint check as a task

* Update CI to run proselint check

* Make changes suggested by proselint

* Fix typo in ci.yml

* Add alex to github actions checks

* Apply changes based on Alex's suggestions

* Include alex check as a task

* Apply alex's suggestion

MichalisPanayides committed:
Merge branch 'add-poa-analysis-of-game' of https://github.com/11michalis11/ED-EMS-game-paper into add-poa-analysis-of-game

MichalisPanayides committed:
Compile main.pdf

MichalisPanayides committed:
Apply changes based on alex and proselint

MichalisPanayides committed:
Add .alexrc.yml file

This is a configuration file for alex so that certain of its suggestions
are ignored. In this case the blocking time (labelled as B_i) was found
as an incensitive phrase.

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
 
-\subsection{Data Analysis}
\ No newline at end of file
+The system defined in section \ref{sec:model_overview} can be directly applied 
```
```suggestion
The system defined in section \ref{sec:model_overview} can be applied 
```

(Less words more betterer)

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+the Emergency Medical Services (EMS) as a distributor that distributes 
+patients to them.
+The parameters of the model can also be mapped into parameters of the ED and 
+the EMS.
```
```suggestion
The parameters of the model correspond to the following parameters of the ED and 
the EMS:
```

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+to a healthcare scenario with two such queueing systems.
+The two queueing systems can be thought of as two Emergency Departments (ED) and 
+the Emergency Medical Services (EMS) as a distributor that distributes 
+patients to them.
```
```suggestion
patients to them.

% TODOAdd a small paragraph here referencing the news articles as anecdotes of the problems this causes.

```

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+    \item \( C_i \): The number of available resources (beds) in ED \(i\)  
+    \item \( T_i \): The strategy that ED \(i\) chooses to play
+    \item \( N_i \): The total patient capacity of ED \(i\)
+    \item \( M_i \): The total parking capacity of ED \(i\)
```
```suggestion
    \item \( \lambda_{1_i} \): The arrival rate of other patients to ED \(i\in\{1, 2\}\)
    \item \( \mu_i \): The service rate of patients at ED \(i\in\{1, 2\}\)
    \item \( C_i \): The number of available resources (beds) in ED \(i\in\{1, 2\}\)  
    \item \( T_i \): The action that ED \(i\in\{1, 2\\) chooses to play which corresponds to the threshold at which they do not accept EMS patients.
    \item \( N_i \): The total patient capacity of ED \(i\in\{1, 2\\)
    \item \( M_i \): The total parking capacity of ED \(i\in\{1, 2\\)
```

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+    \item \( N_i \): The total patient capacity of ED \(i\)
+    \item \( M_i \): The total parking capacity of ED \(i\)
+    \item \( R \): The time target for both EDs
+    \item \( \alpha \in [0, 1] \) : \textit{Importance} of blocking time and 
```
```suggestion
    \item \( \alpha \in [0, 1] \) : \textit{Weighting} of blocking time and 
```

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+    lost individuals (equation \ref{eq:obj-distributor})
+\end{itemize}
+
+The patients that are distributed by the EMS arrive at the hospital via an 
+ambulance and are then either unloaded at the ED or stay blocked outside in 
+the ambulance.
+Whether or not the ambulance and the patient stay blocked is determined by 
+the threshold that the given ED chooses to play.
+High threshold indicates that the ED accepts patients even if it is relatively 
+full, while low threshold means that the ED blocks ambulances more frequently.
```
Would this paragraph be better BEFORE the list of parameters?

MichalisPanayides replied:
Hmmm. Yeah I agree. It makes a bit more sense to include it before

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+High threshold indicates that the ED accepts patients even if it is relatively 
+full, while low threshold means that the ED blocks ambulances more frequently.
+
+The game to be played now is between the two EDs and the EMS. 
```
```suggestion
```

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+
+The game to be played now is between the two EDs and the EMS. 
+The strategies of the two EDs are the range of thresholds that they can choose
+from and their utilities is the proportion of individuals whose time in the 
+system is within a predetermined target time.
+The EMS has to decide how to distribute its patients among the two EDs so that 
+the blocking time of ambulances is minimised. 
```
Should this not incorporate the weighting/importance (`alpha`). So should this say something like:

```
their utilities are a weighted combination of the proportion of 
individuals whose time in the system is within a predetermined 
target **and** the percentage of lost patients?
```

MichalisPanayides replied:
Yep, I see what you are saying. BUT, `alpha` is used for the EMS decision (i.e. what percentage to send to each hospital). So I think I should include something like that:

```suggestion
from and their utilities is the proportion of individuals whose time in the 
system is within a predetermined target time.
The EMS has to decide how to distribute its patients among the two EDs so that 
the weighted combination of the ambulance blocking time and the percentage of 
lost ambulances is minimised. 
```

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+is a Normal form game that is then used to inform the decision of the EMS.
+Note that the formulated game here assumes that prior to making a choice the 
+EMS knows the strategies that each ED is playing (figure 
+\ref{fig:imperfect-info-game}).
```
```suggestion
Note that the formulated game here assumes that prior to making a choice the 
EMS knows the strategies that each ED is playing (figure 
\ref{fig:imperfect-info-game}). This corresponds to reacting to experienced delays.
```

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+EMS knows the strategies that each ED is playing (figure 
+\ref{fig:imperfect-info-game}).
+
+\subsection{Data Analysis}
```
```suggestion
\subsection{Numerical Results}
```

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+asymmetric replicator dynamics is used \cite{asymmetricreplicatordynamics}.
+The use of a learning algorithm allows to investigate, not only the outcome of 
+the game, but also how that outcome is reached.
+Consider figure \ref{fig:ard-by-itself}. 
```
Before moving to the replicator dynamics include a small numerical example where you compute the NE.

Then discuss what that means but also discuss how we are not sure if NE will naturally be reached.

Then something like: 

```
... since the work of Maynard Smith <cite> EGT gives tools for the emergence 
of stable behaviour.
One such model that allows for asymetric payoffs, as is the case above,
is replicator dynamics ...
Stable outcomes of this algorithm will correspond to a subset of Nash
equilibria but more importantly
give a model of emergent behaviour. ... 
```

drvinceknight replied:
We then also need to include some of the mathematics.

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+Consider figure \ref{fig:ard-by-itself}. 
+By running asymmetric replicator dynamics on the system the behaviour that 
+emerges can be observed.
+It can be seen that for these particular set of parameters the strategies of 
```
```suggestion
It can be seen that for this particular set of parameters the strategies of 
```

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+    \end{itemize}
+\end{multicols}
+
+To study the efficiency of the strategies being played a new concept is 
+introduced that considers the ratio between each hospital's best achievable 
+blocking time and the one that is being played.
```
It's not 100% new, we need to include some discussion of the literature on PoA here and explain how your compartmentalised one is different (which you do below).

drvinceknight replied:
Once you do that we might need to tweak some of the rest.

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+blocking time and the one that is being played.
+This new concept is defined as the compartmentalised price of anarchy of the 
+players of the game and is defined as \(PoA_i(\tilde{s})\), where 
+\(i \in [1, 2]\) to distinguish among the two players/hospitals where 
```
```suggestion
\(i \in \{1, 2\}\) to distinguish among the two players/hospitals where 
```

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+    PoA_{i}(\tilde{s}) = \frac{B_i(\tilde{s})}{\min_{s \in S_i} B_i(s)}
+\end{equation}
+
+Consider a game with the following parameters:
```
Add a sentence specifying how this differs from the other set of parameters.

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+    \end{itemize}
+\end{multicols}
+
+\textbf{Base run:}
```
```suggestion
\textbf{Initial scenario:}
```

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+
+\textbf{Base run:}
+Using equation (\ref{eq:poa_compartmentalised}) and asymmetric replicator 
+dynamics the game can be simulated and show the compartmentalised price of 
```
```suggestion
dynamics the emergent behaviour can be measured and the compartmentalised price of 
```

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+    \label{fig:ard_original}
+\end{figure}
+
+It can be observed that the learning algorithm reaches a stable pair of 
```
```suggestion
The learning algorithm reaches a stable pair of 
```

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+
+\textbf{Increasing \(\lambda_2\):}
+Figure \ref{fig:ard_lambda_2} shows a similar run of the
+algorithm but when the strategies begin to stabilise a huge increase in the
```
```suggestion
algorithm but when the strategies begin to stabilise an increase in the
```

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+By increasing \(\lambda_2\) there is no change as to how players behave
+(\(T_1 = 5, T_2 = 6\)), but the efficiency of the system does change. 
+There is a decline in the price of anarchy of the blocking time which at first 
+glance indicates that upon flooding the system it becomes more efficient. 
+This is non-sensical though.
```
```suggestion
glance indicates that upon flooding the system it becomes the 
loss in efficiency due to rational individual behaviour decreases.
```

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+glance indicates that upon flooding the system it becomes more efficient. 
+This is non-sensical though.
+What it really shows is that the steep increase in \( \lambda_2 \) leaves 
+less potential for improvement. 
```
```suggestion
the system unable to cope regardless of the decisions made.
```

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+the previous cases.
+Once the played strategies in asymmetric replicator dynamics strategies 
+converge the payoff matrices of the two are scaled in such a way so that the 
+utilities of the selected strategy is penalised.
```
```suggestion
utilities of the selected strategy are penalised. This corresponds to a precise
policy change where more societally beneficial behaviours are incentivised.
```

drvinceknight commented on sections/healthcare_application/main.tex:
```diff
@@ -2,4 +2,240 @@ \section{EMS - ED application}
+    }
+    \label{fig:ard_penalty}
+\end{figure}
+
+Figure \ref{fig:ard_penalty} shows that players start playing strategies 
+\( T_1 = 5\) and \( T_2 = 6 \) and mid-run of the learning algorithm a penalty
+is applied to these strategies on the payoff matrix.
+By incentivising the players in such a way the players change their strategies 
```
We should probably include the actual matrices here?

drvinceknight reviewed (changes requested):
Looks good @11michalis11, some overall suggestions. Happy to jump in a call on Friday to discuss anything if that would be helpful.

MichalisPanayides committed:
Apply suggestions from code review

Co-authored-by: Vince Knight <vincent.knight@gmail.com>

MichalisPanayides committed:
Apply minor changes

MichalisPanayides committed:
Update known words for spellchecker

MichalisPanayides committed:
More changes form code review and fix minor typos

MichalisPanayides committed:
Add small paragraph of news articles

MichalisPanayides committed:
Add more around analysing games

- Include short Nash equilibrium paragraph and why we will not use it
- Include small numeric example of NE
- Include mathematics around asymmetric replicator dynamics

MichalisPanayides committed:
Apply proselint suggestion

MichalisPanayides committed:
Add dicussion for price of anarchy

MichalisPanayides committed:
Include payoff matrices

MichalisPanayides committed:
Update known words

MichalisPanayides committed:
Fix typo

MichalisPanayides committed:
Apply changes based on supervisory meeting

drvinceknight reviewed (approved):
Looks good to me @11michalis11, no comments.
